include/nbl/video/IGPUCommandPool.h

```diff
@@ -4,38 +4,803 @@
+        : ICommand(calc_size(queryPool, accelerationStructureCount, accelerationStructures)), m_queryPool(core::smart_refctd_ptr<const IQueryPool>(queryPool)), m_accelerationStructureCount(accelerationStructureCount)
+    {
+        m_accelerationStructures = new (this + sizeof(CWriteAccelerationStructurePropertiesCmd)) core::smart_refctd_ptr<const IGPUAccelerationStructure>[m_accelerationStructureCount];
+
+        for (auto i = 0; i < m_accelerationStructureCount; ++i)
+            m_accelerationStructures[i] = core::smart_refctd_ptr<const IGPUAccelerationStructure>(accelerationStructures[i]);
+    }
+
+    ~CWriteAccelerationStructurePropertiesCmd()
+    {
+        for (auto i = 0; i < m_accelerationStructureCount; ++i)
+            m_accelerationStructures[i].~smart_refctd_ptr();
+    }
+
+    static uint32_t calc_size(const IQueryPool* queryPool, const uint32_t accelerationStructureCount, IGPUAccelerationStructure const *const *const accelerationStructures)
+    {
+        return core::alignUp(sizeof(CWriteAccelerationStructurePropertiesCmd) + (accelerationStructureCount + 1)* sizeof(void*), alignof(ICommand));
+    }
+
+private:
+    core::smart_refctd_ptr<const IQueryPool> m_queryPool;
+    const uint32_t m_accelerationStructureCount;
+    core::smart_refctd_ptr<const IGPUAccelerationStructure>* m_accelerationStructures;
+};
+
+class IGPUCommandPool::CBuildAccelerationStructuresCmd : public IGPUCommandPool::ICommand
+{
+public:
+    CBuildAccelerationStructuresCmd(const uint32_t accelerationStructureCount, core::smart_refctd_ptr<const IGPUAccelerationStructure>* accelerationStructures, const uint32_t bufferCount, core::smart_refctd_ptr<const IGPUBuffer>* buffers)
+        : ICommand(calc_size(accelerationStructureCount, accelerationStructures, bufferCount, buffers)), m_resourceCount(accelerationStructureCount + bufferCount)
+    {
+        m_resources = new (this + sizeof(CWriteAccelerationStructurePropertiesCmd)) core::smart_refctd_ptr<const IReferenceCounted>[m_resourceCount];
+
+        uint32_t k = 0u;
+        for (auto i = 0; i < accelerationStructureCount; ++i)
+            m_resources[k++] = core::smart_refctd_ptr<const IReferenceCounted>(accelerationStructures[i]);
+
+        for (auto i = 0; i < bufferCount; ++i)
+            m_resources[k++] = core::smart_refctd_ptr<const IReferenceCounted>(buffers[i]);
+    }
+
+    ~CBuildAccelerationStructuresCmd()
+    {
+        for (auto i = 0; i < m_resourceCount; ++i)
+            m_resources[i].~smart_refctd_ptr();
+    }
+
+    static uint32_t calc_size(const uint32_t accelerationStructureCount, core::smart_refctd_ptr<const IGPUAccelerationStructure>* accelerationStructures, const uint32_t bufferCount, core::smart_refctd_ptr<const IGPUBuffer>* buffers)
+    {
+        const auto resourceCount = accelerationStructureCount + bufferCount;
+        return core::alignUp(sizeof(CBuildAccelerationStructuresCmd) + resourceCount * sizeof(void*), alignof(ICommand));
```

Comment:
use `sizeof(core::smart_refctd_ptr<const IReferenceCounted>)` and `alignof(CBuildAccelerationStructuresCmd)` for clarity here

Reply:
Resolved.